xbmc/addons/Addon.cpp

```diff
@@ -32,57 +32,93 @@
 
 using XFILE::CDirectory;
 using XFILE::CFile;
+using namespace KODI::ADDONS;
 
 namespace ADDON
 {
 
 CAddon::CAddon(const AddonInfoPtr& addonInfo, TYPE addonType)
   : m_addonInfo(addonInfo),
-    m_userSettingsPath(),
-    m_loadSettingsFailed(false),
-    m_hasUserSettings(false),
     m_profilePath(StringUtils::Format("special://profile/addon_data/{}/", m_addonInfo->ID())),
-    m_settings(nullptr),
     m_type(addonType == ADDON_UNKNOWN ? addonInfo->MainType() : addonType)
 {
-  m_userSettingsPath = URIUtils::AddFileToFolder(m_profilePath, "settings.xml");
 }
 
 /**
  * Settings Handling
  */
-bool CAddon::HasSettings()
+
+std::vector<uint32_t> CAddon::GetKnownInstanceIds() const
 {
-  return LoadSettings(false) && m_settings->HasSettings();
+  return m_addonInfo->GetKnownInstanceIds();
 }
 
-bool CAddon::SettingsInitialized() const
+bool CAddon::SupportInstanceSettings() const
 {
-  return m_settings != nullptr && m_settings->IsInitialized();
+  return m_addonInfo->SupportInstanceSettings();
+}
+
+bool CAddon::DeleteInstanceSettings(uint32_t instance)
+{
+  const auto itr = m_settings.find(instance);
+  if (instance == KODI::ADDONS::ADDON_SETTINGS_ID || itr == m_settings.end())
+    return false;
```

Comment:
If KODI::ADDONS::ADDON_SETTINGS_ID is passed here you can prevent the call to `find`:

```c++
if (instance == KODI::ADDONS::ADDON_SETTINGS_ID)
  return false;

const auto itr = m_settings.find(instance);
if (itr == m_settings.end())
  return false;